src/Networks/Rutanio/Rutanio/Networks/RutanioMain.cs

```diff
@@ -0,0 +1,285 @@
+
+            Bech32Encoders = new Bech32Encoder[2];
+            var encoder = new Bech32Encoder(network.CoinTicker.ToLowerInvariant());
+            Bech32Encoders[(int)Bech32Type.WITNESS_PUBKEY_ADDRESS] = encoder;
+            Bech32Encoders[(int)Bech32Type.WITNESS_SCRIPT_ADDRESS] = encoder;
+
+            Checkpoints = network.Checkpoints;
+            DNSSeeds = network.DNS.Select(dns => new DNSSeedData(dns, dns)).ToList();
+            SeedNodes = network.Nodes.Select(node => new NBitcoin.Protocol.NetworkAddress(IPAddress.Parse(Dns.GetHostAddresses(node).GetValue(0).ToString()), network.DefaultPort)).ToList();
+
+            StandardScriptsRegistry = new RutanioStandardScriptsRegistry();
+
+            // 64 below should be changed to TargetSpacingSeconds when we move that field.
+            Assert(DefaultBanTimeSeconds <= Consensus.MaxReorgLength * 64 / 2);
+
+            Assert(Consensus.HashGenesisBlock == uint256.Parse(network.HashGenesisBlock));
+            Assert(Genesis.Header.HashMerkleRoot == uint256.Parse(network.HashMerkleRoot));
+
+            RegisterRules(Consensus);
+            RegisterMempoolRules(Consensus);
+        }
+
+        protected void RegisterRules(IConsensus consensus)
+        {
+            consensus.ConsensusRules
+                .Register<HeaderTimeChecksRule>()
+                .Register<HeaderTimeChecksPosRule>()
+                .Register<RutanioBugFixPosFutureDriftRule>()
+                .Register<CheckDifficultyPosRule>()
+                .Register<RutanioHeaderVersionRule>()
+                .Register<ProvenHeaderSizeRule>()
+                .Register<ProvenHeaderCoinstakeRule>();
+
+            consensus.ConsensusRules
+                .Register<BlockMerkleRootRule>()
+                .Register<PosBlockSignatureRepresentationRule>()
+                .Register<PosBlockSignatureRule>();
+
+            consensus.ConsensusRules
+                .Register<SetActivationDeploymentsPartialValidationRule>()
+                .Register<PosTimeMaskRule>()
+
+                // rules that are inside the method ContextualCheckBlock
+                .Register<TransactionLocktimeActivationRule>()
+                .Register<CoinbaseHeightActivationRule>()
+                .Register<WitnessCommitmentsRule>()
+                .Register<BlockSizeRule>()
+
+                // rules that are inside the method CheckBlock
+                .Register<EnsureCoinbaseRule>()
+                .Register<CheckPowTransactionRule>()
+                .Register<CheckPosTransactionRule>()
+                .Register<CheckSigOpsRule>()
+                .Register<PosCoinstakeRule>();
+
+            consensus.ConsensusRules
+                .Register<SetActivationDeploymentsFullValidationRule>()
+
+                .Register<CheckDifficultyHybridRule>()
+
+                /// rules that require the store to be loaded (coinview)
+#pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
Use the new rules for coindb (see BTC chain)

Reply:
We have updated to the new rules for coindb